pkgs/tools/misc/swaglyrics/default.nix

```
@@ -0,0 +1,44 @@
+{ stdenv, lib, python3, fetchFromGitHub, ncurses }:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "swaglyrics";
+  version = "1.2.2";
+
+  src = fetchFromGitHub {
+    owner = "SwagLyrics";
+    repo = "SwagLyrics-For-Spotify";
+    rev = "v${version}";
+    sha256 = "1dwj9fpyhqqpm2z3imp8hfribkzxya891shh77yg77rc2xghp7mh";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    unidecode colorama beautifulsoup4 flask requests swspotify
+  ];
+
+  preConfigure = ''
+    substituteInPlace setup.py \
+      --replace 'requests>=2.24.0' 'requests~=2.23'
+  '';
+
+  preBuild = "export HOME=$NIX_BUILD_TOP";
+
+  # Skip tests that require network access
+  patches = [ ./skip_network_test.patch ];
+  postPatch = ''
+     sed -i '/test_database_for_unsupported_song/i\\	@unittest.skip\("No network access during checkPhase"\)' tests/test_cli.py
+     sed -i '/test_that_lyrics_works_for_unsupported_songs/i\\	@unittest.skip\("No network access during checkPhase"\)' tests/test_cli.py
+     sed -i '/test_that_get_lyrics_works/i\\	@unittest.skip\("No network access during checkPhase"\)' tests/test_cli.py
+  '';
```

since you're using pytestCheckHook, this should work
```suggestion
  # disable tests which touch network
  disabledTests = [
     "test_database_for_unsupported_song"
     "test_that_lyrics_works_for_unsupported_songs"
     "test_that_get_lyrics_works"
  '';
```